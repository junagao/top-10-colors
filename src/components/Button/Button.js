import React from 'react'
import { bool, func } from 'prop-types'
import ButtonStyled from './Button.styles'

const Button = ({ isRandomRating, startRandomRating, stopRandomRating }) =>
  isRandomRating ? (
    <ButtonStyled type="button" onClick={stopRandomRating}>
      stop
    </ButtonStyled>
  ) : (
    <ButtonStyled type="button" onClick={startRandomRating}>
      random rate
    </ButtonStyled>
  )

Button.propTypes = {
  isRandomRating: bool.isRequired,
  startRandomRating: func.isRequired,
  stopRandomRating: func.isRequired,
}

export default Button
