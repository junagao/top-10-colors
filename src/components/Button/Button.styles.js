import styled from '@emotion/styled'
import theme from 'theme'

const ButtonStyled = styled.button`
  color: ${({ reset }) => (reset ? '#FF4D4F' : theme.colors.text)};
  cursor: pointer;
  border-radius: ${theme.radii.base};
  border: 0;
  border: ${({ reset }) =>
    reset ? `1px solid #FF4D4F` : `1px solid ${theme.colors.secondary}`};
  font-size: ${theme.fontSizes.sm};
  margin-left: ${theme.space[4]};
  padding: ${theme.space[4]};
  outline: none;
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
  &:hover {
    color: ${({ reset }) => (reset ? '#FF6E70' : theme.colors.light)};
    background-color: ${({ reset }) =>
      reset ? '#fff' : theme.colors.secondary};
    border: ${({ reset }) =>
      reset ? `1px solid #FF6E70` : `1px solid ${theme.colors.primary}`};
  }
`

export default ButtonStyled
