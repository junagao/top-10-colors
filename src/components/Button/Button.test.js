import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Button', () => {
  let wrapper
  const mockText = 'Rate'
  const onClickMock = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Button text={mockText} reset={false} onClick={onClickMock} />,
    )
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one ButtonStyled element of type `button` with prop `reset`', () => {
    expect(wrapper.find('ButtonStyled').length).toEqual(1)
    expect(wrapper.find('ButtonStyled').prop('type')).toEqual('button')
    expect(wrapper.find('ButtonStyled').prop('reset')).toBeFalsy()
    wrapper.setProps({ reset: true })
    expect(wrapper.find('ButtonStyled').prop('reset')).toBeTruthy()
  })

  it('should render the button text', () => {
    expect(wrapper.find('ButtonStyled').children().text()).toEqual('Rate')
  })

  it('should test click event on button and call onClick one time', () => {
    wrapper.find('ButtonStyled').simulate('click')
    expect(onClickMock.mock.calls.length).toEqual(1)
  })
})
