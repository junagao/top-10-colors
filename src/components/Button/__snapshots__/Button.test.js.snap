// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly given the required props 1`] = `
<ButtonStyled
  onClick={[MockFunction]}
  reset={false}
  type="button"
>
  Rate
</ButtonStyled>
`;
